pkg/reconciler/testing/mongodbsource.go

```diff
@@ -84,7 +84,7 @@ func WithMongoDbSourceConnectionFailed(err string) MongoDbSourceOption {
 }
 
 // WithMongoDbSourceConnectionSuccess updates the status of the connection to be successful.
-func WithMongoDbSourceConnectionSuccess(err string) MongoDbSourceOption {
+func WithMongoDbSourceConnectionSuccess() MongoDbSourceOption {
```

Comment:
I think if you are not passing any argument, you don't need to return an option... 
just call s.Status.MarkConnectionSuccess() in the body and that's it...
you can do that in the follow ups